[Rust] Add `LowLevelILSSARegister` type

emesare opened the pull request:
The partial register id is the "main" id so we return that, whereas the full register id is not.

@bdash did you have any more thoughts surrounding the usage of this API?

emesare committed:
[Rust] Add `LowLevelILSSARegisterKind::id` helper

The partial register id is the "main" id so we return that, whereas the full register id is not.

bdash commented:
I'm not sure about this API. It does not seem obvious to a client what `id` will return for a partial register. Maybe a doc comment fixes that, or perhaps there's a clearer name.

I'm not even 100% sure whether `get_ssa_register_uses` and friends are doing the right thing with how they handle partial registers at the moment.

emesare committed:
[Rust] Add `LowLevelILSSARegister` to correctly refer to the full ssa register when querying

emesare commented:
> I'm not even 100% sure whether `get_ssa_register_uses` and friends are doing the right thing with how they handle partial registers at the moment.

Yea it seems that it was never intended to be queried with a partial register, I fixed up the `LowLevelILSSARegisterKind` to make it so that partial register will not be able to be passed to any of those api's. I also added some more doc comments to hopefully try and explain what `Full` and `Partial` ssa register kinds are.

https://github.com/Vector35/binaryninja-api/pull/7745/commits/11cc1662ab4bbd364e46b29e7afcfdadacb702ea

emesare committed:
[Rust] Fix impl `Display` for `LowLevelILRegisterKind`
